File: car/sensorColor.py
import time
from machine import Pin, ADC
import neopixel

class SensorColor:
    def __init__ (self, pinADC, pinDG):
    	self.red = 0
    	self.green = 0
    	self.blue = 0

    	# self.sensorLDR = ADC(Pin(pinADC))
        self.sensorLDR = ADC(pinADC)
    	# self.sensorLDR.atten(ADC.ATTN_11DB)
    	self.sensorNP = neopixel.NeoPixel(Pin(pinDG), 1)
    


    def changeColor(self, color):
    	if (color == 'G'):
    		self.sensorNP[0] = (0, 255, 0) #green
    	elif (color == 'R'): 
    		self.sensorNP[0] = (255, 0, 0) #red
    	elif (color == 'B'): 
    		self.sensorNP[0] = (0, 0, 255) #blue
    	self.sensorNP.write()

    def readSensor(self):
    	self.changeColor('R')
    	time.sleep_ms(5)
    	self.red = self.sensorLDR.read()
    	# print('red -------------', self.red)

    	self.changeColor('G')
    	time.sleep_ms(5)
    	self.green = self.sensorLDR.read()
    	# print('green ----------------------', self.green)

    	self.changeColor('B')
    	time.sleep_ms(5)
    	self.blue = self.sensorLDR.read()
    	# print('blue ----------------------------', self.blue)

    	if (self.red > self.green and self.red > self.blue):
    		return b"VERMELHO"
    		# print("------------ VERMELHO ----------")
        elif (self.green > self.red and self.green > self.blue
            and self.red>900 and self.green>900  and self.blue>800):
            return b"BRANCO"
    	elif (self.green > self.red and self.green > self.blue):
    		return b"VERDE"
    		# return "------------ VERDE ----------"
    	elif(self.blue > self.red and self.blue > self.green):
    		return b"AZUL"
    		# return "------------ AZUL ----------"
    	elif(self.red < self.green and self.green > self.blue):
    		return b"PRETO"
    		# return "------------ PRETO ----------"
    	else:
    		return b"NDA"
    		# return "------------ NDA ----------"
			

File: car/sensorColor.c
#include <Adafruit_NeoPixel.h>
#ifdef _AVR_
#include <avr/power.h> // Required for 16 MHz Adafruit Trinket
#endif
#define PIN        A1
#define NUMPIXELS 1
#define COLOR_DELAY 5

Adafruit_NeoPixel pixels(NUMPIXELS, PIN, NEO_GRB + NEO_KHZ800);


void setup() {
#if defined(_AVR_ATtiny85_) && (F_CPU == 16000000)
  clock_prescale_set(clock_div_1);
#endif
  pixels.begin(); // INITIALIZE NeoPixel strip object (REQUIRED)
  Serial.begin(115200);
  pinMode(2,OUTPUT);
  pinMode(3,OUTPUT);
}

void changeColor(char color) {
  pixels.clear();
  for (int i = 0; i < NUMPIXELS; i++) {
    if (color == 'G') pixels.setPixelColor(i, pixels.Color(0, 255, 0));
    else if (color == 'R') pixels.setPixelColor(i, pixels.Color(255, 0, 0));
    else if (color == 'B') pixels.setPixelColor(i, pixels.Color(0, 0, 255));
  }
  pixels.show();
}

int red, green, blue;
void readSensor(){
  
  changeColor('R');
  delay(COLOR_DELAY);
  red = analogRead(A0);


  changeColor('G');
  delay(COLOR_DELAY);
  green = analogRead(A0);

  changeColor('B');
  delay(COLOR_DELAY);
  blue = analogRead(A0);

  
//  Serial.print("\n Vermelho: ");
//  Serial.print(red);
//  Serial.print("\n Verde: ");
//  Serial.print(green);
//  Serial.print("\n Azul: ");
//  Serial.print(blue);

  changeColor(' ');
//
  if (red > green && red > blue) Serial.print("------------ VERMELHO ----------");
  else if (green > red && green > blue) Serial.print("------------ VERDE ----------");
  else if (blue > red && blue > green) Serial.print("------------ AZUL ----------");
  else Serial.println("\n------------ NDA ----------");
  delay(50);
}

void loop() {
  Serial.println("sensor 1");
  digitalWrite(2,HIGH);
  digitalWrite(3,LOW);
  readSensor();
  delay(10);
  Serial.print("sensor 2");
  digitalWrite(2,LOW);
  digitalWrite(3,HIGH);
  readSensor();
  delay(100); 
}